PROTO_tests/tests/message_test.py

```diff
@@ -0,0 +1,234 @@
+            == json.dumps(msg.asdict())
+        )
+
+        # self.assertFalse(self.assertRaises(Exception, json.loads(json.dumps(msg.asdict()))))
+
+        assert json.loads(json.dumps(msg.asdict())) == msg.asdict()
+
+        min_msg = RequestMessage(user="user1", cmd="connect")
+
+        assert (
+            "RequestMessage(user='user1', token=None, cmd='connect', format=STRING, args=None, size=-1)"
+            == str(min_msg)
+        )
+
+        assert (
+            "RequestMessage(user='user1', token=None, cmd='connect', format=STRING, args=None, size=-1)"
+            == repr(min_msg)
+        )
+
+        assert (
+            '{"user": "user1", "token": "", "cmd": "connect", "format": "STRING", "args": "", "size": -1}'
+            == json.dumps(min_msg.asdict())
+        )
+
+    def test_reply_msg(self):
+        msg = ReplyMessage(msg="normal result", msgType=MessageType.NORMAL, user="user")
+        msgDupe = ReplyMessage(msg="normal result", msgType=MessageType.NORMAL, user="user")
+        msgNonDupe = ReplyMessage(msg="normal result 2", msgType=MessageType.NORMAL, user="user")
+
+        assert msg == msgDupe
+        assert msg != msgNonDupe
+
+        assert "ReplyMessage(msg='normal result', msgType=NORMAL, user='user')" == str(msg)
+        assert "ReplyMessage(msg='normal result', msgType=NORMAL, user='user')" == repr(msg)
+
+        newMsg = ReplyMessage.fromdict({"msg": "normal result", "msgType": "NORMAL", "user": "user"})
+        assert msg == newMsg
+
+        assert "ReplyMessage(msg='normal result', msgType=NORMAL, user='user')" == str(newMsg)
+        assert "ReplyMessage(msg='normal result', msgType=NORMAL, user='user')" == repr(newMsg)
+
+        with pytest.raises(ValueError):
+            ReplyMessage.fromdict({"msg": "normal result", "msgType": "NORMAL"})
+
+
+class TestJSONArgs:
+    # TODO numpy dtypes are not supported by json, we probably want to add an issue to handle this
+    SCALAR_TYPES = [int, float, bool, str]
+    PDA_TYPES = [ak.dtype(t) for t in ak.DTypes if t != "str"]
+    @pytest.mark.parametrize("dtype", SCALAR_TYPES)
+    def test_scalar_args(self, dtype):
+        val1 = dtype(5)
+        val2 = dtype(0)
+        size, args = _json_args_to_str({"arg1": val1, "arg2": val2})
+        expected = json.dumps([
+            json.dumps({"key": "arg1", "objType": "VALUE", "dtype": ak.resolve_scalar_dtype(val1), "val": str(val1)}),
+            json.dumps({"key": "arg2", "objType": "VALUE", "dtype": ak.resolve_scalar_dtype(val2), "val": str(val2)})
+        ])
+        assert args == expected
+
+    def test_addl_str(self):
+        val = "abc"
+        size, args = _json_args_to_str({"arg": val})
+        expected = json.dumps([
+            json.dumps({"key": "arg", "objType": "VALUE", "dtype": ak.resolve_scalar_dtype(val), "val": val}),
+        ])
+        assert args == expected
+
+    @pytest.mark.parametrize("dtype", [int])
+    def test_list_arg(self, dtype):
```

Comment:
should we do this test for dtypes other than just `int`? if not, i feel like maybe don't parameterize it and just use `int`?

Reply:
This should us the `SCALAR_TYPES`. I initially built it with `int` and forgot to update once everything was done. It should be updated now.